File: environment-spring-boot/casadocodigo/src/main/java/org/ecorp/casadocodigo/repositories/EstadoRepository.java
package org.ecorp.casadocodigo.repositories;

import java.util.List;
import org.ecorp.casadocodigo.model.Estado;
import org.ecorp.casadocodigo.model.Pais;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface EstadoRepository extends JpaRepository<Estado, Long> {

  List<Estado> findByPais(Pais pais);

}

File: src/main/kotlin/com/tiozao/cdd/loja/repository/EstadoRepository.kt
package com.tiozao.cdd.loja.repository

import com.tiozao.cdd.loja.repository.entity.EstadoEntity
import org.springframework.data.repository.CrudRepository
import org.springframework.stereotype.Repository

@Repository
interface EstadoRepository : CrudRepository<EstadoEntity, Int>{
}